### ISSUE TITLE
`options` removed from Money 6.13.5

### ISSUE THREAD

**Issue by prognostikos** [RubyMoney/eu_central_bank#101]
Seems that this gem isn't compatible with the latest money version (6.13.5)

```
NameError: undefined local variable or method `options' for #<Money::RatesStore::StoreWithHistoricalDataSupport:0x000055ad13572018>
Did you mean?  @options
/home/circleci/mrclean/vendor/bundle/ruby/2.6.0/gems/eu_central_bank-1.4.2/lib/money/rates_store/store_with_historical_data_support.rb:23:in `transaction'
/home/circleci/mrclean/vendor/bundle/ruby/2.6.0/gems/eu_central_bank-1.4.2/lib/eu_central_bank.rb:179:in `update_parsed_rates'
/home/circleci/mrclean/vendor/bundle/ruby/2.6.0/gems/eu_central_bank-1.4.2/lib/eu_central_bank.rb:29:in `update_rates'
/home/circleci/mrclean/app/better_now/accounting/ecb_bank.rb:25:in `update_rates'
/home/circleci/mrclean/app/better_now/accounting/ecb_bank.rb:20:in `initialize'
/home/circleci/mrclean/app/better_now/accounting/ecb_bank.rb:12:in `new'
/home/circleci/mrclean/app/better_now/accounting/ecb_bank.rb:12:in `setup'
/home/circleci/mrclean/config/initializers/money_exchange_bank.rb:4:in `<top (********)>'
```

This looks like the change that causes the problem: https://github.com/RubyMoney/money/pull/898/files#diff-70409fffc5690f30fb7495abab7a7dbfL103

**Comment by ioquatix** [comment]
https://github.com/RubyMoney/eu_central_bank/blob/b413b1c3dfbbc0146cbaa4de935eed8e8fc6fbd3/lib/money/rates_store/store_with_historical_data_support.rb#L13-L33 is horribly broken.

```ruby
    # Wraps block execution in a thread-safe transaction
    def transaction(force_sync = false, &block)
      # Ruby 1.9.3 does not support @mutex.owned?
      if @mutex.respond_to?(:owned?)
        force_sync = false if @mutex.locked? && @mutex.owned?
      else
        # If we allowed this in Ruby 1.9.3, it might possibly cause recursive
        # locking within the same thread.
        force_sync = false
      end
      if !force_sync && (@in_transaction || options[:without_mutex])
        block.call self
      else
        @mutex.synchronize do
          @in_transaction = true
          result = block.call
          @in_transaction = false
          result
        end
      end
    end
```

As soon as `@in_transaction = true` is set, any other thread can enter in the block without synchronisation.

If `result = block.call` raises an exception, `@in_transaction = false` is never called.

**Comment by shailesh-kalamkar** [comment]
@ioquatix It would be helpful to know if you found any alternative.

**Comment by ioquatix** [comment]
@shailesh-kalamkar An alternative to what?

**Comment by shailesh-kalamkar** [comment]
@ioquatix An alternative to this gem which is using the same API. Basically, would like to know how did you overcome the issue?

Thanks.

**Comment by ioquatix** [comment]
@shailesh-kalamkar https://github.com/ioquatix/latinum

**Comment by shailesh-kalamkar** [comment]
Thank you so much.

**Comment by antstorm** [comment]
Sorry about this, 6.13.6 is now available that include a fix for this regression

**Comment by juleswelsch** [comment]
https://github.com/RubyMoney/money/issues/900#issuecomment-562070633

**Comment by antstorm** [comment]
Sorry again, just published [eu_central_bank 1.5.0](https://rubygems.org/gems/eu_central_bank/versions/1.5.0) that should finally fix this and rely on a thread-safe implementation of the superclass for synchronisation

**Comment by ioquatix** [comment]
@antstorm great work!
